crypto/s2n_libcrypto.c

```diff
@@ -125,6 +125,19 @@ bool s2n_libcrypto_is_openssl(void)
     return !is_other_libcrypto_variant;
 }
 
+/* Check if the linked libcrypto is a fips-capable openssl variant.
+ *
+ * This method does not check whether the linked libcrypto has fips mode enabled.
+ */
+bool s2n_libcrypto_is_openssl_fips(void)
```

Comment:
This was also added in https://github.com/aws/s2n-tls/pull/5081, but [with a different meaning](https://github.com/aws/s2n-tls/pull/5081/files#diff-d57cb85db6d68617fc2dd29256dfee929d486b006fe49f60561c7eea6db53837R128-R132).

This also won't return true for openssl3-fips (and we probably wouldn't want it to for this use case). To avoid having two similar APIs here, would it maybe make sense to just add a `#elif` to `s2n_supports_custom_rand` and return false specifically for OpenSSL 1.0.2-fips? I'm not sure it's going to be that useful anywhere else anyway.